core/src/main/scala/akka/kafka/internal/TransactionalProducerStage.scala

```diff
@@ -146,8 +163,10 @@ private final class TransactionalProducerStageLogic[K, V, P](stage: Transactiona
 
   override val onMessageAckCb: AsyncCallback[Envelope[K, V, P]] =
     getAsyncCallback[Envelope[K, V, P]](_.passThrough match {
-      case o: ConsumerMessage.PartitionOffset => batchOffsets = batchOffsets.updated(o)
-      case _ =>
+      case o: ConsumerMessage.PartitionOffsetCommittedMarker =>
+        batchOffsets = batchOffsets.updated(o)
+        println(s"ADDING OFF ${o.offset}")
+      case x => println("Chuj " + x)
```

Comment:
I think your frustration might be caused by another bug with offset tracking. With a fix for that applied, your test case does not fail for me anymore. The fix is this one: https://github.com/akka/alpakka-kafka/pull/752/commits/d381b4f451e76f125a39424d62ea243eebbae0f3

Reply:
Sorry, this should not be included in the PR. Will rewrite history.